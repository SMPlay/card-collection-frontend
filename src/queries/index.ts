export { GET_CARD_COLLECTIONS } from "./get-card-collections";
export { GET_COLLECTIONS_NAME } from "./get-collections-name";
export { LOGIN } from "./login";
export { IS_AUTH } from "./is-auth";
export { GET_REFRESH_TOKEN } from './get-refresh-token';
export { REGISTRATION_QUERY } from "./registration-query";
export { CONFIRM_USER } from "./confirm-user";
