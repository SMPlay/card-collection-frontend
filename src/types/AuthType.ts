export interface AuthType<T> {
  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  handleChange: (e: React.ChangeEvent<any>) => void;
  handleBlur: (e: any) => void;
  values: T;
  status: string;
  errors: T;
  touched: T;
}
