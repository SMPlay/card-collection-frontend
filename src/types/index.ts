export { CollectionType } from './CollectionType'
export { PageType } from './PageType'