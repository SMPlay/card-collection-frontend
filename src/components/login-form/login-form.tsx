import React from "react";
import { TextField, Box, Button, Typography } from "@material-ui/core";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import { Link } from "react-router-dom";

import { FormikType } from "../../types/FormikType";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      "& .MuiTextField-root": {
        margin: theme.spacing(1),
        width: "60ch",
      },
    },
    loginField: {
      display: "flex",
      flexDirection: "column",
    },
    buttonsField: {
      display: "flex",
      flexDirection: "row",
      alignItems: "center",
    },
    navLink: {
      marginLeft: 10,
      color: "#000",
      opacity: 1,
      transition: "opacity .2s ease",
      "&:hover": {
        opacity: 0.8,
      },
    },
  })
);

export interface Values {
  login?: string | boolean;
  password?: string | boolean;
}

export const LoginForm: React.FC<FormikType<Values>> = ({
  handleSubmit,
  handleChange,
  values,
  status,
  errors,
}) => {
  const styles = useStyles();

  return (
    <form onSubmit={handleSubmit} className={styles.root}>
      {status?.error ? (
        <Typography variant="h4" component="h2">
          {status.error}
        </Typography>
      ) : (
        ""
      )}
      <Box m={2} className={styles.loginField}>
        <TextField
          id="login"
          value={values.login}
          onChange={handleChange}
          label="Логин"
          placeholder="Введите логин"
          variant="outlined"
          error={errors.login || status?.error ? true : false}
          helperText={errors.login}
          required
          disabled={status?.loading}
        />
        <TextField
          id="password"
          value={values.password}
          onChange={handleChange}
          label="Пароль"
          placeholder="Введите пароль"
          variant="outlined"
          type="password"
          error={errors.password || status?.error ? true : false}
          helperText={errors.password}
          required
          disabled={status?.loading}
        />
      </Box>
      <Box className={styles.buttonsField}>
        <Button
          variant="contained"
          color="primary"
          type="submit"
          disabled={status?.loading}
        >
          Войти
        </Button>
        <Link className={styles.navLink} to="/reset-password">
          Забыли пароль?
        </Link>
      </Box>
    </form>
  );
};
