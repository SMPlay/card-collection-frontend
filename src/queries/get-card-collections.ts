import { gql } from "@apollo/client";

export const GET_CARD_COLLECTIONS = gql`
  query {
    cardCollections {
      id name imageUrl cardsCount release
    }
  }
`;
