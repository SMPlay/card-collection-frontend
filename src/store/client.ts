import { ApolloClient } from "@apollo/client";
import { createUploadLink } from "apollo-upload-client";

import { IS_AUTH } from "../queries";

import { cache } from "./cache";

const link = createUploadLink({ 
  uri: 'http://localhost:4001/graphql',
  credentials: "include"
});

export const client = new ApolloClient({
  link,
  cache,
});

client.writeQuery({
  query: IS_AUTH,
  data: {
    isAuth: false
  }
});
