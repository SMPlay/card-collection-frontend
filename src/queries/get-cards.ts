import { gql } from "@apollo/client";

export const GET_CARDS = gql`
    query cards($from: Int!, $limit: Int!, $collectionName: String!){
        cards(from: $from, limit: $limit, collectionName: $collectionName){
            id
            name
            imageUrl
            role
            rarity
            need
            have
            number

        }

    }
`;
