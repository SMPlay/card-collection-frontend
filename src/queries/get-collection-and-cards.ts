import { gql } from "@apollo/client";

export const GET_COLLECTION_AND_CARDS = gql`
    query cardCollection($id: ID!){
        cardCollection(id: $id){
            id
            name
            cardCollectionName
            imageUrl
            cardsCount
            release
            cards{
                id
                name
                imageUrl
                role
                rarity
                need
                have
                number
            }  
        }
    }
`;
