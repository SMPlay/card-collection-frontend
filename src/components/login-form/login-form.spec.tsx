import React from "react";
import { Typography } from "@material-ui/core";
import { create } from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import { BrowserRouter as Router } from "react-router-dom";
import { mount } from "enzyme";

import { LoginForm, Values } from "./login-form";
import { LOGIN } from "../../queries";
import { AuthType } from "../../types/AuthType";

describe("login form component", () => {
  let props: AuthType<Values>;
  const mocksInitial = [
    {
      request: {
        query: LOGIN,
        variables: { login: "test", password: "test" },
      },
      result: {
        data: {
          user: { id: "123" }
        },
      },
    }
  ];

  beforeEach(() => {
    props = {
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      values: {
        login: "test",
        password: "test"
      },
      status: {
        loading: false,
        error: ""
      },
      errors: {
        login: "",
        password: ""
      },
    }
  });

  it("should render", () => {
    const tree = create(
      <MockedProvider mocks={mocksInitial}>
        <Router>
          <LoginForm {...props}/>
        </Router>
      </MockedProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should get error", () => {
    const component = mount(
      <MockedProvider mocks={mocksInitial}>
        <Router>
          <LoginForm 
            {...props}
            status={{loading: false, error: "Неправильный логин и/или пароль"}}
          />
        </Router>
      </MockedProvider>
    );

    expect(component.find(Typography).text()).toBe("Неправильный логин и/или пароль")
  });
});
