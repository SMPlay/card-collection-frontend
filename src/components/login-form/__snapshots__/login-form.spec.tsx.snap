// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form component should render 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[MockFunction]}
>
  
  <div
    className="MuiBox-root MuiBox-root-5 makeStyles-loginField-2"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="login"
        id="login-label"
      >
        Логин
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="login"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Введите логин"
          required={true}
          type="text"
          value="test"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
          >
            <span>
              Логин
               *
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="password"
        id="password-label"
      >
        Пароль
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Введите пароль"
          required={true}
          type="password"
          value="test"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
          >
            <span>
              Пароль
               *
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-10 makeStyles-buttonsField-3"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Войти
      </span>
    </button>
    <a
      className="makeStyles-navLink-4"
      href="/reset-password"
      onClick={[Function]}
    >
      Забыли пароль?
    </a>
  </div>
</form>
`;
